neuron/src/app/Neuron/Config.hs

```diff
@@ -50,8 +52,22 @@ parsePure :: forall a. FromDhall a => FilePath -> Text -> Either String a
 parsePure fn s = do
   expr0 <- first show $ Dhall.Parser.exprFromText fn s
   expr <- maybeToRight "Cannot have imports" $ traverse (const Nothing) expr0
-  void $ first show $ Dhall.TypeCheck.typeOf expr
-  first show $
-    validationToEither $
-      Dhall.extract @a Dhall.auto $
-        Dhall.Core.normalize expr
+  Dhall.Core.Record (Dhall.Map.keysSet -> fields) <- first show $ Dhall.TypeCheck.typeOf expr
+  let spuriousFields = fields \\ configKeys
+  case null spuriousFields of
+    True ->
+      first show $
+        validationToEither $
+          Dhall.extract @a Dhall.auto $
+            Dhall.Core.normalize expr
+    False -> do
+      let spuriousMessage =
+            "Configuration in neuron.dhall includes spurious fields: " <>
+            intercalate ", " (("`" <>) . (<> "`") . toString <$> toList spuriousFields) <>
+            "."
+          -- TODO: remove once
+          mathJaxMessage =
+            if "mathJaxSupport" `member` spuriousFields
+              then " To disable MathJax support, include a (possibly blank) `./head.html` file."
+              else ""
+      Left $ spuriousMessage <> mathJaxMessage
```

Comment:
1. There should be a comment explaining what this piece of code does.

1. If spuriousFields is a list of length 1 containing just "mathJaxSupport" - it should be ignored, without returning an error. For other cases, it should error out. The idea is that we remove `mathJaxSupport` entirely, while still allowing its presence in older neuron.dhall (if only to ignore it). Running the new neuron on an older zettelkasten shouldn't fail.

Reply:
Okay, I'll add an explanation and a special case for when `mathJaxSupport` is the only spurious field.

(Also, the "TODO: remove once" comment was meant to continue: ... "mathJaxSupport has been deprecated for long enough")